### PR TITLE
Automatically add all other versions of Gandalf to the Replaces section 

### PR DESCRIPTION
**Pull request by jonisc**
Few things I want to tell you:

* In line 62 and 83 we need the "$". It's an anchor for the end of the line. So if there are spaces at the end of a line in all_versions.txt the regular expression would not match. That's why there's line 60. It removes all spaces at the end of a line in that file. 

----
Technically it works like this: in the beginning build.sh checks if the version (its identifier e.g. io.github.ethanrdoesmc.gandalf102) we are currently building is already added in all_versions.txt, if not, it'll add it:

`if [ "$(cat all_versions.txt | grep ${PKG_PACKAGE_REGEX}$)" = "" ]; then
  echo "Adding currently building version to all_versions.txt..."
  printf "${PKG_PACKAGE}\n" >> all_versions.txt
fi`

This file is a list of all Gandalf versions. build.sh will grep every version except the one we are currently building (that's the -v) and add it to a temporary file in $TEMPDIR/gandalf.

`OTHER_VERSIONS=$(cat all_versions.txt | grep -v ${PKG_PACKAGE_REGEX}'$')

\# Save them in a temporary file
printf "${OTHER_VERSIONS}\n" > $GDN_TEMPDIR/tmp_replaces.tmp
\# Add all packages which are in replaces.txt to the temporary file
cat $1/replaces.txt >> $GDN_TEMPDIR/tmp_replaces.tmp
\# Sort them, parse them in one line and save them in "PKG_REPLACES"

PKG_REPLACES=$(cat $GDN_TEMPDIR/tmp_replaces.tmp | sort | sed ':a;N;$!ba;s/\n/,\ /g')`

Here we have the problem that if we have a package called 
io.github.ethanrdoesmc.gandalf102
and a package called io.github.ethanrdoesmc.gandalf102**4** our regular expression `io\.github\.ethanrdoesmc\.gandalf102` (Defined here: `PKG_PACKAGE_REGEX="io\.github\.ethanrdoesmc\.gandalf$1"`)
would match both, and both won't be added to the temporary file. So we need the "$" anchor at the end of our regex. Now it will only match the exact identifier. After we must make sure that there are no characters like a space " " at the end of a line. If there is one, our regex won't match (because at the end of the line there's a space and not the last character of our wanted identifier) 

So we must remove all spaces at the end of a line in all_versions.txt:

`sed -i 's/ *$//' all_versions.txt`

Now we add all other identifiers of Replaces.txt to this file, sort it and add it to the variable PKG_REPLACES. Then we can add its content to the Debian CONTROL file…

I hope that I haven't forgotten anything important.
Ah, I'll have to explain this:

`if [ "$TMPDIR" == "" ]; then
  echo NOTE: Variable '$TMPDIR' is empty! This is not an error. If you"'"re building on iOS this is normal. Make directory...
  GDN_TEMPDIR="/tmp/gandalf"
  mkdir ${GDN_TEMPDIR}

  else 
  GDN_TEMPDIR="$TMPDIR/gandalf"
  mkdir $GDN_TEMPDIR

fi`
On iOS $TMPDIR is empty (or I have made a typo??). So we need to manually declare a temporary directory.

### CODE REVIEW

File: Gandalf/all_versions.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+io.github.ethanrdoesmc.gandalfx
+io.github.ethanrdoesmc.gandalf102
```
Comment (jonisc) [inline Gandalf/all_versions.txt]: Last line must be free.

**Comment by jonisc** [comment]
@EthanRDoesMC and @mehulrao
Please test this on macOS. I only tested it on iOS.

**Comment by EthanRDoesMC** [comment]
Booting Mac VM, hang on

**Comment by jonisc** [comment]
On my Linux box it works…
But haven't tested the .deb yet. 
One thing I want to tell you: I have created a local .gitignore, so git won't upload .debs to GitHub. That's why there are no more debs in Gandalf/ hopefully I haven't removed them from the repo…

**Comment by jonisc** [comment]
@EthanRDoesMC:

Now you can skip this by adding allow-multiple-installs after the version:
`./build.sh x allow-multiple-installs` now will allow you to compile Gandalf without the script adding all other versions to the Replaces section. It'll now only add the content of replaces.txt to the Replaces section. This can be useful if you want to test multiple gandalf versions on one device. 
This is not 100% tested by me, but it should work ;-)

**Comment by EthanRDoesMC** [comment]
nice. mergin' now.

**Pull request merged**